### PR TITLE
code to fix inclue mailin library

### PR DESCRIPTION
**Pull request by amarpanday**
code to fix inclue mailin library

### CODE REVIEW

File: CronResponce.php [hunk 1]
```diff
@@ -46,9 +47,6 @@
     }
     
     $handle = fopen(_PS_MODULE_DIR_ . 'sendinblue/csv/SyncToSendinblue.csv', 'w');
-    $key_value = array();
-    $key_value[] = '';
-    fputcsv($handle, $key_value, '');
```
Comment (Quetzacoalt91) [inline CronResponce.php]: By removing these lines, you do write anymore in the file. Why do you still open then ?

File: EmptyImportOldOrdersFile.php [hunk 2]
```diff
@@ -32,9 +32,6 @@
 
 if (Tools::getValue('proc_success') != '') {
     $handle = fopen(_PS_MODULE_DIR_ . 'sendinblue/csv/ImportOldOrdersToSendinblue.csv', 'w');
-    $key_value = array();
-    $key_value[] = '';
-    fputcsv($handle, $key_value, '');
```
Comment (Quetzacoalt91) [inline EmptyImportOldOrdersFile.php]: Same thing about the files only opened/closed

File: EmptyImportSubUsersFile.php [hunk 3]
```diff
@@ -32,9 +32,6 @@
 
 if (Tools::getValue('proc_success') != '') {
     $handle = fopen(_PS_MODULE_DIR_ . 'sendinblue/csv/ImportSubUsersToSendinblue.csv', 'w');
-    $key_value = array();
-    $key_value[] = '';
-    fputcsv($handle, $key_value, '');
```
Comment (Quetzacoalt91) [inline EmptyImportSubUsersFile.php]: Same thing here too

File: ajaxsubscribeconfig.php [hunk 4]
```diff
@@ -32,4 +33,15 @@
 
 $id_shop_group = Tools::getValue('id_shop_group', 'NULL');
 $id_shop = Tools::getValue('id_shop', 'NULL');
-Configuration::updateValue('Sendin_Subscribe_Setting', Tools::getValue('managesubscribe'), '', $id_shop_group, $id_shop);
+$manage_subs_val = Tools::getValue('managesubscribe');
+$sendin = new Sendinblue();
+$resp = $sendin->defaultNlStatus();
+
+if ($resp == 1) {
+    Configuration::updateValue('Sendin_Subscribe_Setting', $manage_subs_val, '', $id_shop_group, $id_shop);
```
Comment (Quetzacoalt91) [inline ajaxsubscribeconfig.php]: Because they are exactly the same, this line could be moved before the test.

File: sendinblue.php [hunk 5]
```diff
@@ -2605,6 +2669,24 @@ public function subscribeByruntime($email, $post_value = '')
...
+        if (version_compare(_PS_VERSION_, '1.5.3.4', '>') && isset($id_shop) && !empty($email)) {
+            if ($id_shop === null) {
+                $id_shop = 1;
+            }
+            $all_group_reg = array();
+            $all_group_unsubs = array();
+            $all_group_reg = Db::getInstance()->getRow('SELECT GROUP_CONCAT(id_shop) as storeid FROM ' . _DB_PREFIX_ . 'customer  WHERE email = "' . pSQL($email) . '" GROUP BY email');
+            
+            $all_group_unsubs = Db::getInstance()->getRow('SELECT GROUP_CONCAT(id_shop) as storeid FROM ' . _DB_PREFIX_ . 'sendin_newsletter  WHERE email = "' . pSQL($email) . '" GROUP BY email');
+            $data_first_store = array();
```
Comment (Quetzacoalt91) [inline sendinblue.php]: Be careful, you initialize multiple times the same variable.

Ther first time with an `array()`, then with a function result.

File: sendinblue.php [hunk 6]
```diff
@@ -2605,6 +2669,24 @@ public function subscribeByruntime($email, $post_value = '')
...
+            $data_second_store = array();
```
Comment (Quetzacoalt91) [inline sendinblue.php]: Same comment here about the variable initialization

File: sendinblue.php [hunk 7]
```diff
@@ -2605,6 +2669,24 @@ public function subscribeByruntime($email, $post_value = '')
             $attribute_data[] = $value_group;
             $attribute_key[] = 'GROUP_ID';
         }
+        if (version_compare(_PS_VERSION_, '1.5.3.4', '>') && isset($id_shop) && !empty($email)) {
+            if ($id_shop === null) {
+                $id_shop = 1;
+            }
+            $all_group_reg = array();
+            $all_group_unsubs = array();
```
Comment (Quetzacoalt91) [inline sendinblue.php]: These two variable do not need a initialisation with `array()`.

File: sendinblue.php [hunk 8]
```diff
@@ -2739,14 +2830,46 @@ public function subscribeByruntimeRegister($email, $id_gender, $fname, $lname, $
...
             $attribute_key[] = 'GROUP_ID';
         }
-        
+        if (version_compare(_PS_VERSION_, '1.5.3.4', '>') && isset($id_shop) && !empty($email)) {
+            if ($id_shop === null) {
+                $id_shop = 1;
+            }
+            
+            $all_group_reg = array();
+            $all_group_unsubs = array();
```
Comment (Quetzacoalt91) [inline sendinblue.php]: Here again, they get their actual value with `DB::getInstance(...)`. You do not need to initialize them with `array()`

File: sendinblue.php [hunk 9]
```diff
@@ -2832,6 +2957,59 @@ public function codeDeTracking()
...
+        ></span></div>';
+
+        $sendin_smtp_detail = Configuration::get('Sendin_Smtp_Result', '', $this->id_shop_group, $this->id_shop);
+        $smtp_data = Tools::jsondecode($sendin_smtp_detail);
+        $sendin_smtp = !empty($smtp_data->result->relay_data->status) ? $smtp_data->result->relay_data->status : '';
+        $smtp_alert = '';
+        if ($sendin_smtp !== 'enabled') {
+            $smtp_alert = '<div class="alert '. $this->cl_version .'"> ' . $this->l('You need an active SMTP (transactional) account to be able to send confirmation emails. Please').' <a href="mailto:contact@sendinblue.com">' . $this->l('contact customer service').'</a> ' . $this->l('to activate it.').'</div>';
+        }
+
```
Comment (Quetzacoalt91) [inline sendinblue.php]: In order to follow the MVC architecture, HTML code should be generated in a tpl file, not in a PHP controller. You will see that your code in this file will be greatly lighter. :wink:

Comment (amarpanday) [inline sendinblue.php]: I have added html code in tpl  file.

File: CronResponce.php (hunk 1 above)
Comment (amarpanday) [inline CronResponce.php]: This code is used to write blank data after processing as we don't want any once to access these data by using some URL.

File: EmptyImportOldOrdersFile.php (hunk 2 above)
Comment (amarpanday) [inline EmptyImportOldOrdersFile.php]: This code is used to write blank data after processing as we don't want any once to access these data by using some URL.

File: EmptyImportSubUsersFile.php (hunk 3 above)
Comment (amarpanday) [inline EmptyImportSubUsersFile.php]: This code is used to write blank data after processing as we don't want any once to access these data by using some URL.

File: sendinblue.php (hunk 5 above)
Comment (amarpanday) [inline sendinblue.php]: yes i have fixed and changed variable name

File: sendinblue.php (hunk 6 above)
Comment (amarpanday) [inline sendinblue.php]: yes i have fixed and changed variable name

File: sendinblue.php (hunk 7 above)
Comment (amarpanday) [inline sendinblue.php]: yes i have fixed and changed variable name

File: sendinblue.php (hunk 8 above)
Comment (amarpanday) [inline sendinblue.php]: yes i have fixed and changed variable name

File: ajaxsubscribeconfig.php (hunk 4 above)
Comment (amarpanday) [inline ajaxsubscribeconfig.php]: ok I have done

**Comment by amarpanday** [comment]
Hi Quetzacoalt91 

Please check the code and merge it. 

Thanks

**Comment by Quetzacoalt91** [comment]
Hi @amarpanday,

Thanks, your PR has just been merged.
Please also note that you have been granted write access to the repository for the next releases. :wink: 

Best regards

**Pull request merged**